### PR TITLE
Use dupes lookup for host_subject_id

### PR DESCRIPTION
**Pull request by squirrelo**
Requires [american-gut-web #542](https://github.com/biocore/american-gut-web/pull/542)

Uses dupes lookup table to standardize participant names for host_subject_id creation

NOTE: this will fail until #542 is merged. Restart the tests to make it not fail.

### CODE REVIEW

**Comment by mortonjt** [comment]
Would it be worthwhile to add a test case to the data_access tests?

**Comment by squirrelo** [comment]
we don’t really have data access tests for this currently. Also, that would requirere factoring and tests for the entirety of the pulldown code which, while planned, is a logistical nightmare outside the scope of this pul request.

**Comment by mortonjt** [comment]
I'm not comfortable merging this without tests.  Any thoughts @wasade, @ElDeveloper @josenavas ?

File: knimin/lib/data_access.py [hunk 1]
```diff
@@ -679,9 +685,11 @@ def format_survey_data(self, md):  # noqa
             md[1][barcode]['BODY_PRODUCT'] = md_lookup[site]['BODY_PRODUCT']
             md[1][barcode]['DESCRIPTION'] = md_lookup[site]['DESCRIPTION']
             md[1][barcode]['HOST_COMMON_NAME'] = md_lookup[site]['COMMON_NAME']
+
+            participant_name = dupes_lookup.get(
+                md[1][barcode]['SURVEY_ID'], specific_info['participant_name'])
             md[1][barcode]['HOST_SUBJECT_ID'] = sha512(
-                specific_info['ag_login_id'] +
-                specific_info['participant_name']).hexdigest()
+                specific_info['ag_login_id'] + participant_name).hexdigest()
```
Comment (wasade) [inline knimin/lib/data_access.py]: is the name, email, etc lowered?

Comment (squirrelo) [inline knimin/lib/data_access.py]: Nope, name was not lowercased. Fixed.

Comment (squirrelo) [inline knimin/lib/data_access.py]: This shouldn't affect things though, as the case-sensitive names are also in the duplicates table.

**Comment by squirrelo** [comment]
Comment addressed

**Comment by wasade** [comment]
@mortonjt, more formal testing is being added in but as @squirrelo points out, there is quite a bit of logistics to go through and realistically will take a few weeks. Unit testing is not the only way to test code, but it is of course a powerful way.

**Pull request merged**